2_process.smk

```diff
@@ -0,0 +1,155 @@
+        'sequences_file': lake_sequence_files_input,
+        'num_sequences': sequence_counts
+    })
+    df_counts.to_csv(summary_file, index=False)
+
+
+def dynamic_filenames(site_id, file_category = 'dynamic_mntoha'):
+    """
+    Return the files that contain dynamic data that are needed to construct
+    sequences for a given lake.
+
+    This function also triggers four checkpoints:
+    1. fetch_mntoha_metadata to get lake_metadata.csv
+    2. unzip_archive for this lake's drivers
+    3. unzip_archive for this lake's clarity
+    3. unzip_archive for this lake's ice flags
+
+    :param site_id: NHDHR lake ID
+    :param file_category: category of files, e.g., dynamic_mntoha. Used by
+    unzip_archive to determine parent directory.
+    :returns: List of 3 dynamic filenames: drivers, clarity, and ice flags
+
+    """
+    # make this function depend on fetch_mntoha_metadata
+    # needed because lake_metadata.csv is used to determine dynamic files
+    lake_metadata_file = checkpoints.fetch_mntoha_metadata.get().output[0]
+    lake_metadata = pd.read_csv(lake_metadata_file)
+    lake = lake_metadata.loc[lake_metadata['site_id']==site_id].iloc[0]
+    # also make this function depend on unzip_archive
+    # needed to link unzipped files with unzip_archive rule
+    drivers_directory = f'inputs_{lake.group_id}'
+    unzip_archive_drivers = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=drivers_directory
+    ).output[0]
+    clarity_directory = f'clarity_{lake.group_id}'
+    unzip_archive_clarity = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=clarity_directory
+    ).output[0]
+    ice_flags_directory = f'ice_flags_{lake.group_id}'
+    unzip_archive_ice_flags = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=ice_flags_directory
+    ).output[0]
+
+    # dynamic filenames
+    drivers_file = f'{unzip_archive_drivers}/{lake.meteo_filename}'
+    clarity_file = f'{unzip_archive_clarity}/gam_{lake.site_id}_clarity.csv'
+    ice_flags_file = f'{unzip_archive_ice_flags}/pb0_{lake.site_id}_ice_flags.csv'
+    return [drivers_file, clarity_file, ice_flags_file]
+
+
+# Summarize training sequences
+rule process_mntoha:
+    input:
+        mntoha_lake_sequence_files
+    output:
+        "2_process/out/mntoha_sequences/mntoha_sequences_summary.csv"
+    run:
+        save_sequences_summary(input, output[0])
+
+
+# Create .npy of input/output sequences for one lake to use for training and testing
+rule mntoha_lake_sequences:
+    input:
+        "2_process/tmp/mntoha/lake_metadata_augmented.csv",
+        "2_process/tmp/mntoha/temperature_observations_interpolated.csv",
+        lambda wildcards: dynamic_filenames(wildcards.site_id, file_category='dynamic_mntoha')
+    output:
+        "2_process/out/mntoha_sequences/sequences_{site_id}.npy"
+    params:
+        config = config
```

Comment:
Wait I'm confused -- I see that the configuration file is referenced here and on line 152 through the variable `config`, but where the variable for `"2_process/process_config.yaml"` is defined on line 5, the variable is named as `configfile`. Does `snakemake` know that config = configfile, or did you maybe switch the variable name you were using at one point?

Reply:
That's the syntax that Snakemake uses for [configuration](https://snakemake.readthedocs.io/en/latest/snakefiles/configuration.html) 🤷🏻.

Reply:
huh - bizarre!